src/core/sync/event.d

```diff
@@ -0,0 +1,318 @@
+/**
+ * The event module provides a primitive for lightweight signalling of other threads
+ * (aka Windows Events)
+ *
+ * Copyright: Copyright (c) 2019 D Language Foundation
+ * License: Distributed under the
+ *    $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0).
+ *    (See accompanying file LICENSE)
+ * Authors: Rainer Schuetze
+ * Source:    $(DRUNTIMESRC core/sync/event.d)
+ */
+module core.sync.event;
+
+version (Windows)
+{
+    import core.sys.windows.basetsd /+: HANDLE +/;
+    import core.sys.windows.winerror /+: WAIT_TIMEOUT +/;
+    import core.sys.windows.winbase /+: CreateEvent, CloseHandle, SetEvent, ResetEvent,
+        WaitForSingleObject, INFINITE, WAIT_OBJECT_0+/;
+}
+else version (Posix)
+{
+    import core.sys.posix.pthread;
+    import core.sys.posix.sys.types;
+}
+else
+{
+    static assert(false, "Platform not supported");
+}
+
+import core.time;
+import core.internal.abort : abort;
+
+/**
+ * This class represents an event. Clients of an event are suspended while waiting
```

Comment:
This is a struct.

Reply:
Ok. It started out as a class, but no longer.